### ISSUE TITLE
Changing from dev to master channel - Execution failed for task ':app:transformDexArchiveWithExternalLibsDexMergerForDebug.'

### ISSUE THREAD

**Issue by BroPet89** [flutter/flutter#27542]
I recently tried to move from Dev channel to Master. Now I am getting this issue:

```
D8: Program type already present: android.support.v4.media.MediaBrowserCompat$ItemReceiver
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':app:transformDexArchiveWithExternalLibsDexMergerForDebug'.
> com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/4.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/5.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/6.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/7.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/8.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/9.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/10.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/11.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/12.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/13.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/14.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/15.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/16.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/17.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/18.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/19.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/20.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/21.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/22.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/23.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/24.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/25.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/26.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/27.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/28.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/29.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/30.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/31.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/32.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/33.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/34.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/35.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/36.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/37.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/38.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/39.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/40.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/41.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/42.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/43.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/44.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/45.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/46.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/47.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/48.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/49.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/50.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/51.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/52.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/53.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/54.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/55.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/56.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/57.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/58.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/59.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/60.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/61.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/62.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/63.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/64.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/65.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/66.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/67.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/68.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/69.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/70.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/71.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/72.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/73.jar, 
/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/74.jar
 Program type already present: android.support.v4.media.MediaBrowserCompat$ItemReceiver
  Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. 
Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 9s
Gradle task assembleDebug failed with exit code 1
```

When I searched around for the error:
`Execution failed for task ':app:transformDexArchiveWithExternalLibsDexMergerForDebug'.` 
It seems to point to issues with Firebase packages, such as #14874, #16286 and #11113. I am **NOT** using any Firebase packages. 

I have tried adding `multiDexEnabled true` to my `app/build.gradle` as suggested in the above issues. I have tried adding 
```android { ... 
defaultConfig { ... 
multiDexEnabled true
       }
 } 

dependencies { ...
implementation 'com.android.support:multidex:1.0.3' 
}``` 

I have done a `Flutter Clean`. 
I even tried installing a fresh version of Flutter, the problem still persists. 
I deleted my `pubspec.lock`. when a new one was generated I changed the bottom lines back to 
```sdks:
    dart: ">=2.1.0-dev.5.0 <3.0.0"
    flutter: ">=0.4.0 <2.0.0"```

I then tried rolling back my commit, I changed back to dev branch and I deleted all the .jar`s in 
`/home/bropet/Desktop/mapl-flutter/build/app/intermediates/transforms/dexBuilder/debug/`. 
The problem still persists and I can't debug or run my app. 

[✓] Flutter (Channel dev, v1.2.0, on Linux, locale en_US.UTF-8)
    • Flutter version 1.2.0 at /home/bropet/src/flutter
    • Framework revision 06b979c4d5 (11 days ago), 2019-01-25 14:27:35 -0500
    • Engine revision 36acd02c94
    • Dart version 2.1.1 (build 2.1.1-dev.3.2 f4afaee422)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /home/bropet/Android/Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /snap/android-studio/70/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[✓] Android Studio (version 3.3)
    • Android Studio at /snap/android-studio/70/android-studio
    • Flutter plugin version 32.0.1
    • Dart plugin version 182.5124
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[✓] Android Studio (version 3.2)
    • Android Studio at /snap/android-studio/69/android-studio
    • Flutter plugin version 31.3.1
    • Dart plugin version 181.5656
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1136-b06)

[✓] VS Code (version 1.30.2)
    • VS Code at /usr/share/code
    • Flutter extension version 2.22.3

[✓] Connected device (1 available)
    • SM A300FU • 46d8d65e • android-arm • Android 6.0.1 (API 23)

**Comment by Daniyar111** [comment]
same error

**Comment by Daniyar111** [comment]
I'm writing with sdk 28.0.0, and I have used map_view plugin, image_picker and other flutter plugins.. in map_view I had there are 28.0.0 and 26.1.0 sdk in this project.. I checked them and couldn't find it, cause each checked build.gradle was 28 sdk.. in image_picker another error.. Does it decide that I use the new sdc version, and the plugins have been written for a long time? and if I wrote with the older sdk, would the plugins work? or how does it work? because other plugins that I used, such as location, google_map, etc work fine
